src/sound_trigger/fingerprint/templates.py

```
@@ -0,0 +1,137 @@
+    Mirrors the project's existing next-to-asset caching (the legacy engine's
+    `.npy` files): the expensive peak/hash/index/verifier computation runs once
+    and is reused across runs. Any cache miss, version mismatch, corruption or
+    IO error (e.g. a read-only asset dir) silently falls back to recomputing.
+    """
+    wav_path = Path(wav_path)
+    cache_path = wav_path.with_name(wav_path.name + ".fpcache")
+    key = _cache_key(wav_path, tuning)
+    try:
+        if cache_path.exists():
+            with open(cache_path, "rb") as handle:
+                blob = pickle.load(handle)
```

Ruff (0.15.15)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)